public/src/inverted-index.js

```diff
@@ -92,63 +106,61 @@ class Index {
     return check;
   }
 
-/**
- * @method getIndex
- *
- * Method return index of files in indices object
- *
- * @param {string} name
- * @returns {object}
- */
+  /**
+  * @method getIndex
+  *
+  * Method return index of files in indices object
+  *
+  * @param {string} name
+  * @returns {object}
+  */
   getIndex(name) {
-    if (name && typeof name === 'string') {
-      return this.indices[name];
-    }
-    return this.indices;
+    return name && typeof name === 'string' ?
+      this.indices[name] : this.indices;
   }
 
-/**
- * @method searchFeedback
- *
- * Takes terms of array and fetch result of each token.
- *
- * @param {array} termsArray
- * @param {object} file
- * @returns {object} searchResults
- */
-  searchFeedback(termsArray, file) {
+
+  /**
+  * @method doSearch
+  *
+  * Takes terms of array and fetch result of each token.
+  *
+  * @param {array} termsArray
+  * @param {object} file
+  * @returns {object} searchResults
+  */
+  doSearch(termsArray, fileIndex) {
     const searchResults = {};
     termsArray.forEach((term, index) => {
-      if (file.hasOwnProperty(term)) {
-        searchResults[termsArray[index]] = Array.from(file[term]);
+      if (fileIndex.hasOwnProperty(term)) {
+        searchResults[termsArray[index]] = Array.from(fileIndex[term]);
       } else {
         searchResults[term] = [];
       }
     });
     return searchResults;
   }
 
-/**
- * @method searchIndex
- *
- * Looks for search terms in created index
- *
- * @param {object} currentFile
- * @param {array} searchTerm
- * @returns {object} searchResults
- */
-  searchIndex(currentFile, ...searchTerm) {
+  /**
+  * @method searchIndex
+  *
+  * Looks for search terms in created index
+  *
+  * @param {object} selectedFile
+  * @param {array} searchTerms
+  * @returns {object} searchResults
+  */
+  searchIndex(selectedFile, ...searchTerms) {
     const searchResults = {};
-    let termsArray = [];
-    termsArray = this.sanitizeInput(searchTerm[0]);
-    if (!currentFile) {
+    const termsArray = this.sanitizeInput(searchTerms[0]);
+    if (!selectedFile) {
```

Comment:
Consider these 2 method calls to `searchIndex`:

```
searchIndex('books.json', 'alice', 'in', 'wonderland');
---
searchIndex('alice', 'in', 'wonderland')
```

Because `selectedFile` _is your first argument_, it's value after the first method call, is `books.json`, a valid String, hence `if (!selectedFile)` would not hold true. In this instance, we intentionally passed a filename in.

In the second method call, `selectedFile` is `alice`, another valid String, yet again `if (!selectedFile)` would not be true because the `selectedFile` value is `Truthy`.

Point being, because the `selectedFile` is the first argument, not passing an actual file name but passing `searchTerms`, would result in the first search term being used as the `selectedFile` name, which is not what you want. What you want is that when a filename whose Index exists is passed, the Index for that file is checked directly, otherwise, search all Indices.